draft-ietf-jsonpath-base.md

```diff
@@ -1095,13 +1095,9 @@ filter-selectors enclosing the filter-selector that is directly
 enclosing the identifier).
 
 A test expression either tests the existence of a node
-designated by an embedded query (see {{extest}}) or tests the
-result of a function expression (see {{fnex}}).
-In the latter case, if the function expression is of type
-`ST(OptionalBooleanType)` (see {{typesys}}), it tests whether the result
-is `true`; if the function expression is of type
-`ST(OptionalNodesType)`, it tests whether the result is different from
-`Nothing`.
+designated by an embedded query (see {{extest}}) or function expression
+(see {{fnex}}).  A function is only valid as a test expression if
+it is of the type `ST(OptionalNodesType)`.
```

Comment:
This is not a good match for match() and search().

Reply:
Yes it is.  It follows perfectly from the return value of `match()` and `search()`: a JSON `true` or `false`, or `Nothing`.  I'm just saying that such a value cannot be used in a `test-expr`... because it's meaningless there.